src/coreclr/vm/prestub.cpp

```diff
@@ -2056,18 +2056,28 @@ extern "C" void* STDCALL ExecuteInterpretedMethod(TransitionBlock* pTransitionBl
     return frames.interpMethodContextFrame.pRetVal;
 }
 
-extern "C" void* STDCALL ExecuteInterpretedMethodWithArgs(TransitionBlock* pTransitionBlock, TADDR byteCodeAddr, int8_t* pArgs, size_t size, void* retBuff)
+extern "C" void ExecuteInterpretedMethodFromUnmanaged(MethodDesc* pMD, int8_t* args, size_t argSize, int8_t* ret)
 {
-    // copy the arguments to the stack
-    if (size > 0 && pArgs != nullptr)
+    _ASSERTE(pMD != NULL);
+
+    InterpByteCodeStart* targetIp = pMD->GetInterpreterCode();
+    if (targetIp == NULL)
     {
-        InterpThreadContext *threadContext = GetInterpThreadContext();
-        int8_t *sp = threadContext->pStackPointer;
+        GCX_PREEMP();
+        (void)pMD->DoPrestub(NULL /* MethodTable */, CallerGCMode::Coop);
```

Comment:
`DoPrestub` is GC triggers. What guarantees that the `args` are reported to the GC here when we come here from CallDescrWorker? It looks like a pre-existing bug on the CallDescrWorker path.

I am not sure whether sharing this with CallDescrWorker path is a good idea. CallDescrWorker needs to protect the arguments or avoid calling DoPrestub with unprotected arguments on stack. UMCO path does not have this concern - arguments coming from unmanaged side are guaranteed to not contain GC references.

Reply:
Perhaps we can fix this by copying the arguments to the stack before this call. The branch below does that copy and I can move that above?